lib/jekyll/jekyll-sitemap.rb

```diff
@@ -8,20 +8,24 @@ class JekyllSitemap < Jekyll::Generator
     # Main plugin action, called by Jekyll-core
     def generate(site)
       @site = site
-      unless sitemap_exists?
-        write
-        @site.keep_files ||= []
-        @site.keep_files << "sitemap.xml"
-      end
+      @site.pages << sitemap unless sitemap_exists?
```

Comment:
This *may* cause a problem for sites that iterate through pages, but if we call it out in the changelog, we should be fine.